src/prefect/agent.py

```diff
@@ -63,7 +63,9 @@ def __init__(
 
     async def update_matched_agent_work_queues(self):
         if self.work_queue_prefix:
-            matched_queues = await self.client.match_work_queues(self.work_queue_prefix)
+            matched_queues = []
+            for prefix in self.work_queue_prefix:
+                matched_queues += await self.client.match_work_queues(prefix)
```

Comment:
I think we'll want to modify `match_work_queues` to accept a list of prefixes, otherwise we'll be unnecessarily polling the server for work queues--in the event we choose to move the matching serverside, accepting multiple prefix patterns should be fairly straightforward as well

Reply:
Definitely a nice catch! Will fix